### PR TITLE
Lint: Fixed lint

### PR DESCRIPTION
**Pull request by alvinwan**
- Whenever AWS fails to connect, add all the details of the error to the log.
- Fixed lint errors of the form "ineffectual assignment to..."

depends on #531

### CODE REVIEW

File: cluster/provider/awsSpot.go [hunk 1]
```diff
@@ -63,7 +64,7 @@ func (clst *amazonCluster) Connect(namespace string) error {
 	clst.namespace = namespace
 
 	if _, err := clst.List(); err != nil {
-		return errors.New("AWS failed to connect")
+		return fmt.Errorf("AWS failed to connect: %v", err)
```
Comment (mmussomele) [inline cluster/provider/awsSpot.go]: This change should be its own commit.

File: cluster/provider/awsSpot.go [hunk 2]
```diff
@@ -513,7 +514,7 @@ func (clst *amazonCluster) SetACLs(acls []string) error {
 
 		if !groupIngressExists {
 			log.Debug("Amazon: Add intragroup ACL")
-			_, err = session.AuthorizeSecurityGroupIngress(
+			_, _ = session.AuthorizeSecurityGroupIngress(
```
Comment (mmussomele) [inline cluster/provider/awsSpot.go]: why not just `session.AuthorizeSecurityGroupIngress(`?

**Comment by ejj** [comment]
I think this patch is obsolete now. If you think it should still be merged, go ahead and re-open it.
